fvtest/porttest/omrsockTest.cpp

```diff
@@ -509,6 +515,41 @@ TEST(PortSockTest, two_socket_stream_communication)
 	omrsock_socket_t connectedClientStreamSocket = NULL;
 	EXPECT_EQ(OMRPORTLIB->sock_accept(OMRPORTLIB, serverStreamSocket, &connectedClientStreamSockAddr, &connectedClientStreamSocket), 0);
 
+	char msg[100] = "This is an omrsock test for 2 socket stream communications.";
+	int32_t bytesLeft = strlen(msg) + 1;
+	uint8_t *cursor = (uint8_t *)msg;
+	int32_t bytesSent = 0;
+	while(1) {
+		bytesSent = OMRPORTLIB->sock_send(OMRPORTLIB, connectedClientStreamSocket, cursor, bytesLeft, 0);
+		if (bytesSent < 0) {
+			FAIL();
+			break;
+		}
+		bytesLeft -= bytesSent;
+		cursor += bytesSent;
+		if (0 == bytesLeft) {
+			break;
+		}
+	}
+
+	char buf[100] = "";
+	bytesLeft = strlen(msg) + 1;
+	cursor = (uint8_t *)buf;
+	int32_t bytesRecv = 0;
+	while(1) {
+		bytesRecv = OMRPORTLIB->sock_recv(OMRPORTLIB, clientStreamSocket, cursor, bytesLeft, 0);
+		if (bytesRecv < 0) {
+			FAIL();
+			break;
+		}
+		bytesLeft -= bytesRecv;
+		cursor += bytesRecv;
+		if (0 == bytesLeft) {
+			break;
+		}
```

Comment:
This test should be used as the breaking condition for the while loop.

Reply:
So rather than
```c++
while(1) {
  // ...
		if (0 == bytesLeft) {
			break;
		}
}
```

do:

```c++
while (0 < bytesLeft) {
  // ...
}
```

Reply:
Yes that sounds good :+1: I actually just recently saw that too, but I only changed it for the sendtorecvfrom PR.